### PR TITLE
mgmt add azure-resourcemanager-perf

### PR DESCRIPTION
**Pull request by xseeseesee**
Prepare this module for performance tests according to [Writing-Performance-Tests](https://github.com/Azure/azure-sdk-for-java/wiki/Writing-Performance-Tests).

Please note:
1. I expect this part would not be released in maven so just place the version starting 1.0.0-beta.1.
2. For easy authentication, add azure-resourcemanager-test for using AuthFile to setup the client.
3. `AzureResourceManagerTest` is the base class. `ResourceGroupTestBase` is the class could help create/delete resource group when testing resource creation.
4. For now, the added tests as below:
- List subscriptions
- List tenants
- List resource groups
- Create storage accounts
- Create dns zones

Please feel free to suggest if any simple resources/operations could be added now or later.

Additionally, the format of test result would display as below:
```
=== Results ===
Completed 66 operations in a weighted-average of 117.96s (0.56 ops/s, 1.787 s/op)
```

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-perf/pom.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  <dependencies>
+    <dependency>
+      <groupId>com.azure.resourcemanager</groupId>
+      <artifactId>azure-resourcemanager</artifactId>
+      <version>2.0.0-beta.5</version> <!-- {x-version-update;com.azure.resourcemanager:azure-resourcemanager;current} -->
+    </dependency>
+    <dependency>
+      <groupId>com.azure.resourcemanager</groupId>
+      <artifactId>azure-resourcemanager-test</artifactId>
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-perf/pom.xml:34]: Do we need the test framework?

File: sdk/resourcemanager/azure-resourcemanager-perf/src/main/java/com/azure/resourcemanager/perf/core/AzureResourceManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+        String authFilePath = System.getenv("AZURE_AUTH_LOCATION");
+        if (CoreUtils.isNullOrEmpty(authFilePath)) {
+            System.out.println("Environment variable AZURE_AUTH_LOCATION must be set.");
+            System.exit(1);
+        }
+
+        AuthFile authFile = AuthFile.parse(new File(authFilePath));
+        AzureProfile profile = new AzureProfile(authFile.getTenantId(), authFile.getSubscriptionId(), AzureEnvironment.AZURE);
+        azureResourceManager = AzureResourceManager.authenticate(authFile.getCredential(), profile).withDefaultSubscription();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-perf/src/main/java/com/azure/resourcemanager/perf/core/AzureResourceManagerTest.java:30]: How does data-plane prepare their test env?

File: sdk/resourcemanager/azure-resourcemanager-perf/pom.xml (hunk 1 above)
Comment (xccc-msft) [reply sdk/resourcemanager/azure-resourcemanager-perf/pom.xml:34]: AuthFile from azure-resourcemanager-test.
ResourceNamer from azure-core-test.(As per the potential changes may happen on SdkContext.)

We may remove this if we expect to use other credential and SdkContext.

File: sdk/resourcemanager/azure-resourcemanager-perf/src/main/java/com/azure/resourcemanager/perf/core/AzureResourceManagerTest.java (hunk 2 above)
Comment (xccc-msft) [reply sdk/resourcemanager/azure-resourcemanager-perf/src/main/java/com/azure/resourcemanager/perf/core/AzureResourceManagerTest.java:30]: `String connectionString = System.getenv("STORAGE_CONNECTION_STRING");`
They use the connection string.

File: sdk/resourcemanager/azure-resourcemanager-perf/pom.xml (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-perf/pom.xml:34]: `ResourceNamer` is not exactly required. A random GUID will mostly do.

`AuthFile` seems needed unless we choose a different credential.

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-perf/pom.xml (hunk 1 above)
Comment (xccc-msft) [reply sdk/resourcemanager/azure-resourcemanager-perf/pom.xml:34]: I was putting UUID for creating storage account while the service return errors by resource name restriction.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-perf/pom.xml:34]: storage account name is globally unique, and cannot have "-" in it. These requirement is there whatever util you choose, and no util can help if the name generated conflict with any one exists...

**Pull request merged**